test/e2e/upgrade/alert/alert.go

```diff
@@ -54,48 +45,49 @@ func (t *UpgradeTest) Setup(f *framework.Framework) {
 
 // Test checks if any critical alerts are firing.
 func (t *UpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
-	g.By("Checking for critical alerts")
+	g.By("Checking for alerts")
 
-	// Recover current test if it fails so test suite can complete
-	defer g.GinkgoRecover()
+	start := time.Now()
 
 	// Block until upgrade is done
-	g.By("Waiting for upgrade to finish before checking for critical alerts")
+	g.By("Waiting for upgrade to finish before checking for alerts")
 	<-done
 
-	ctx, cancel := context.WithCancel(context.Background())
-
 	// Additonal delay after upgrade completion
-	g.By("Waiting before checking for critical alerts")
-	time.Sleep(alertCheckSleep)
-	cancel()
+	g.By("Waiting before checking for alerts")
+	time.Sleep(2 * time.Minute)
 
-	if exutil.TolerateVersionSkewInTests() {
-		e2eskipper.Skipf("Test is disabled to allow cluster components to have different versions, and skewed versions trigger multiple other alerts")
-	}
-	t.oc.SetupProject()
-	ns := t.oc.Namespace()
+	ns := t.oc.SetupNamespace()
 	execPod := exutil.CreateExecPodOrFail(t.oc.AdminKubeClient(), ns, "execpod")
 	defer func() {
-		t.oc.AdminKubeClient().CoreV1().Pods(ns).Delete(ctx, execPod.Name, *metav1.NewDeleteOptions(1))
+		t.oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
 	}()
 
-	// Query to check if Prometheus has been up and running for entire post-upgrade
-	// period by verifying Watchdog alert has been in firing state
-	watchdogQuery := fmt.Sprintf(`count_over_time(ALERTS{alertstate="firing",alertname="Watchdog", severity="none"}[%dm])`, alertCheckSleepMinutes)
+	testDuration := time.Now().Sub(start).Truncate(time.Second)
+
+	// Query to check if Prometheus has been up and running for entire upgrade
+	// period by verifying Watchdog alert has been in firing state. To account for
+	// range start/end clipping, add one to the numerator and ensure we are greater
+	// than or equal to one.
+	watchdogQuery := fmt.Sprintf(`((count_over_time(ALERTS{alertstate="firing",alertname="Watchdog",severity="none"}[%s:1s])+1) / %d) >= 1`, testDuration.String(), int(testDuration.Seconds()))
```

Comment:
I don't think the `.String()` is necessary for `%s` formatting (here and a few lines below).